docs/general/xs/functions.json

```diff
@@ -1335,6 +1335,87 @@
+        },
+        {
+            "name": "xsTask",
+            "return_type": "void",
+            "params": [
+                {
+                    "name": "masterUnitID",
+                    "type": "int",
+                    "required": true,
+                    "desc": "Unit to insert the task to."
+                },
+                {
+                    "name": "actionType",
+                    "type": "int",
+                    "required": true,
+                    "desc": "Task type. Eg.: 105 for heal, 155 for aura and etc. Look in the A.G.E."
+                },
+                {
+                    "name": "targetMasterUnitID",
+                    "type": "int",
+                    "required": false,
+                    "desc": "Target unitId for the task if exists. Values 9xx refer to classes."
+                },
+                {
+                    "name": "playerID",
+                    "type": "int",
+                    "required": false,
+                    "desc": "The player to whose units the task will be inserted. If unspecified or -1, applies to all players except Gaia."
+                }
+            ],
+            "desc": "This function inserts or overwrites a task with the fields defined by the `xsTaskAmount` to a specified unit. If the task with the `actionType`, `targetMasterUnitID` and `Search Wait Time` (from `xsTaskAmount`) already exists it is overwritten, otherwise a new one is inserted at the end of the task list."
+        },
+        {
+            "name": "xsRemoveTask",
+            "return_type": "void",
+            "params": [
+                {
+                    "name": "masterUnitID",
+                    "type": "int",
+                    "required": true,
+                    "desc": "Unit to remove the task from."
+                },
+                {
+                    "name": "actionType",
+                    "type": "int",
+                    "required": true,
+                    "desc": "Task type. Eg.: 105 for heal, 155 for aura and etc. Look in the A.G.E."
+                },
+                {
+                    "name": "targetMasterUnitID",
+                    "type": "int",
+                    "required": false,
+                    "desc": "Target unitId for the task if exists. Values 9xx refer to classes."
+                },
+                {
+                    "name": "playerID",
+                    "type": "int",
+                    "required": false,
+                    "desc": "The player from whose units the task will be removed. If unspecified or -1, applies to all players except Gaia."
+                }
+            ],
+            "desc": "This function removes a task from a specified unit. The task is removed if it matches `actionType`, `targetMasterUnitID` and `Search Wait Time` (from `xsTaskAmount`). Other `xsTaskAmount` fields play no role in task removal."
```

Comment:
Consistency:

```
"Removes a task from a unit if the specified `actionType`, `unitId`, and `Search Wait Time` (set by `xsTaskAmount`) match an existing task in a unit. No other fields are used for filtering (same as when `xsTask` edits instead of adding a new task)"
```